app/App/Panel/Notify/AddToken/index.js

```diff
@@ -47,131 +63,152 @@ class AddToken extends React.Component {
-                    if (e.target.value === '') this.setState({ decimals: this.decimalsDefault })
-                  }}
-                />
+                <label className='tokenInputLabel'>
+                  Decimals
+                  <input
+                    className={`tokenInput tokenInputAddress ${this.isDefault('decimals') ? 'tokenInputDim' : ''}`}
+                    value={this.state.decimals} spellCheck='false'
+                    onChange={(e) => {
+                      if (!e.target.value) return this.setState({ decimals: '' })
+                      if (e.target.value.length > 2) return e.preventDefault()
+
+                      const decimals = parseInt(e.target.value)
+                      if (!Number.isInteger(decimals)) return e.preventDefault()
+
+                      this.setState({ decimals })
+                    }}
+                    onFocus={(e) => {
+                      if (e.target.value === this.decimalsDefault) this.setState({ decimals: '' })
+                    }}
+                    onBlur={(e) => {
+                      if (e.target.value === '') this.setState({ decimals: this.decimalsDefault })
+                    }}
+                  />
+                </label>
+
               </div>
 
               <div className='tokenChainId'>
-                <div className='tokenInputLabel'>Chain ID</div>
-                <input
-                  className={this.state.chainId === this.chainDefault ? 'tokenInput tokenInputDim' : 'tokenInput'}
-                  value={this.state.chainId} spellCheck='false'
-                  onChange={(e) => {
-                    if (!e.target.value) return this.setState({ chainId: '' })
-
-                    const chainId = parseInt(e.target.value)
-                    if (!Number.isInteger(chainId)) return e.preventDefault()
-
-                    this.setState({ chainId })
-                  }}
-                  onFocus={(e) => {
-                    if (e.target.value === this.chainDefault) this.setState({ chainId: '' })
-                  }}
-                  onBlur={(e) => {
-                    if (e.target.value === '') this.setState({ chainId: this.chainDefault })
-                  }}
-                />
+                <label className='tokenInputLabel'>
+                  Chain ID
+                  <input
+                    className={`tokenInput tokenInputAddress ${this.isDefault('chainId') ? 'tokenInputDim' : ''}`}
+                    value={this.state.chainId} spellCheck='false'
+                    onChange={async (e) => {
+                      if (!e.target.value) return this.setState({ chainId: '' })
+
+                      const chainId = parseInt(e.target.value)
+                      if (!Number.isInteger(chainId)) {
+                        return e.preventDefault()
+                      }
+
+                      this.setState({ chainId })
+                      await this.updateTokenData(this.state.address, chainId)
```

Comment:
do we need to wait for this to return?

Reply:
Good catch, seems to work without it, so I'll remove. I reckon I kept this in here because I'm used to working around this typescript-eslint rule: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-floating-promises.md

Reply:
yeah, as long as this can't throw any errors that need to be caught we're OK, otherwise we would just need to handle them